android/app/src/main/java/com/mattermost/share/RealPathUtil.java

```diff
@@ -96,19 +96,24 @@ public static String getPathFromSavingTempFile(Context context, final Uri uri) {
         File tmpFile;
         String fileName = null;
 
+        if (uri == null || uri.isRelative()) {
+            return null;
+        }
+
         // Try and get the filename from the Uri
         try {
             Cursor returnCursor =
                     context.getContentResolver().query(uri, null, null, null, null);
             int nameIndex = returnCursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
             returnCursor.moveToFirst();
-            fileName = returnCursor.getString(nameIndex);
+            fileName = sanitizeFilename(returnCursor.getString(nameIndex));
+
         } catch (Exception e) {
             // just continue to get the filename with the last segment of the path
         }
 
         try {
-            if (fileName == null) {
+            if (TextUtils.isEmpty(fileName)) {
                 fileName = uri.getLastPathSegment().toString().trim();
```

Comment:
The Javadoc for [getLastPathSegment](https://developer.android.com/reference/android/net/Uri#getLastPathSegment()) says it decodes the thing it returns -- does that mean `%2F` becomes `/`? If so, we need to call `sanitizeFilename` again here.

Reply:
Answering my own question after a quick test: yes, path separators get decoded so this needs to be sanitized.